server/fleet/app.go

```diff
@@ -788,6 +803,42 @@ func (f *Features) ApplyDefaults() {
 	f.EnableHostUsers = true
 }
 
+// Clone implements cloner for Features.
+func (f *Features) Clone() (interface{}, error) {
+	return f.Copy(), nil
+}
+
+// Copy returns a deep copy of the Features.
+func (f *Features) Copy() *Features {
```

Comment:
not sure if makes much sense to add  this, but in the other PR I wrote some tests for this function

```go
func TestFeaturesCopy(t *testing.T) {
	t.Run("nil receiver", func(t *testing.T) {
		var f *Features
		require.Nil(t, f.Copy())
	})

	t.Run("shallow copy", func(t *testing.T) {
		f := &Features{
			EnableHostUsers:         true,
			EnableSoftwareInventory: false,
		}
		clone := f.Copy()
		require.NotNil(t, clone)
		require.Equal(t, f.EnableHostUsers, clone.EnableHostUsers)
		require.Equal(t, f.EnableSoftwareInventory, clone.EnableSoftwareInventory)
	})

	t.Run("copy AdditionalQueries", func(t *testing.T) {
		rawMessage := json.RawMessage(`{"test": "data"}`)
		f := &Features{
			AdditionalQueries: &rawMessage,
		}
		clone := f.Copy()
		require.NotNil(t, clone.AdditionalQueries)
		require.NotSame(t, f.AdditionalQueries, clone.AdditionalQueries)
		require.Equal(t, *f.AdditionalQueries, *clone.AdditionalQueries)
	})

	t.Run("copy DetailQueryOverrides", func(t *testing.T) {
		f := &Features{
			DetailQueryOverrides: map[string]*string{
				"foo": ptr.String("bar"),
				"baz": nil,
			},
		}
		clone := f.Copy()
		require.NotNil(t, clone.DetailQueryOverrides)
		require.NotSame(t, f.DetailQueryOverrides, clone.DetailQueryOverrides)
		require.Equal(t, f.DetailQueryOverrides["foo"], clone.DetailQueryOverrides["foo"])
		require.NotSame(t, f.DetailQueryOverrides["foo"], clone.DetailQueryOverrides["foo"])
		require.Equal(t, f.DetailQueryOverrides["baz"], clone.DetailQueryOverrides["baz"])
		require.Same(t, f.DetailQueryOverrides["baz"], clone.DetailQueryOverrides["baz"])
	})
}
```

Reply:
Thanks, yeah I think especially the NotSame vs Equal checks are interesting and make it a good place to add checks for new fields that require explicit cloning. I'll copy it into the PR.